sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala

```diff
@@ -626,6 +627,40 @@ private[spark] class HiveExternalCatalog(conf: SparkConf, hadoopConf: Configurat
     client.getPartition(db, table, spec)
   }
 
+  override def listPartitionsByFilter(
```

Comment:
Why can't we just call `client.getPartitionsByFilter` in this method and put all other logic at caller side?